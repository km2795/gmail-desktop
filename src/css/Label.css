.message-label-badge {
  margin: auto 5px;
  font-size: 13px;
  font-family: "Quicksand-Bold";
}

.user-label {
  color: black;
  border: 1px solid rgba(0, 0, 0, 0.2);
}
