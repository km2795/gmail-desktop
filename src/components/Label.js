import React from "react";
import "../css/Label.css";

const Label = ({ color, label, userLabel }) => (
  <span className={`
    badge badge-${color}
    message-label-badge
    ${userLabel ? "user-label" : ""}`
  }>
    {label.replace("CATEGORY_", "")}
  </span>
);

export default Label;
