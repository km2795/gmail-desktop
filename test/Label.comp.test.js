import React from "react";
import renderer from "react-test-renderer";
import Label from "../src/components/Label";

test("should render a user label of a thread", () => {
  const tree = renderer
    .create(<Label 
      color={`white`}
      label={`Archive_Payments`}
      userLabel={true} />)
    .toJSON();
  
  expect(tree).toMatchSnapshot();
});

test("should render a system label of a thread", () => {
  const tree = renderer
    .create(<Label 
      color={`red`}
      label={`INBOX`}
      userLabel={false} />)
    .toJSON();
  
  expect(tree).toMatchSnapshot();
});
