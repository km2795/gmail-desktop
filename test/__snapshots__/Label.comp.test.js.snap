// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a system label of a thread 1`] = `
<span
  className="
    badge badge-red 
    message-label-badge 
    "
>
  INBOX
</span>
`;

exports[`should render a user label of a thread 1`] = `
<span
  className="
    badge badge-white 
    message-label-badge 
    user-label"
>
  Archive_Payments
</span>
`;
